packages/yswaps/scripts/execute-mainnet-pending-trades.ts

```diff
@@ -141,56 +143,90 @@ async function main() {
       }
     );
 
-    const signedTx = await ymech.signTransaction({
-      ...gasParams,
-      type: 2,
-      to: populatedTx.to!,
-      gasLimit: populatedTx.gasLimit!.toNumber(),
-      data: populatedTx.data!,
+    await generateAndSendBundle({
+      pendingTrade,
+      bestSetup,
+      wallet: ymech,
+      executeTx,
+      gasParams,
     });
 
-    console.log('[Execution] Sending transaction in block', await httpProvider.getBlockNumber());
-    const protectTx = await protect.sendTransaction(signedTx);
-    console.log(`[Execution] Transaction submitted via protect rpc - https://protect.flashbots.net/tx/?hash=${protectTx.hash}`);
+    await sleep(DELAY);
+  }
+}
 
-    // const bundle: FlashbotBundle = [
-    //   {
-    //     signedTransaction: signedTx,
-    //   },
-    // ];
-    // console.log('[Execution] Signed with hash:', protectTx.hash);
-    // if (await submitBundle(bundle)) console.log('[Execution] Pending trade', pendingTrade._id, 'executed via', bestSetup.swapper);
+async function generateAndSendBundle(params: {
+  pendingTrade: PendingTrade;
+  bestSetup: TradeSetup;
+  wallet: Wallet;
+  executeTx: PopulatedTransaction;
+  gasParams: {
+    maxFeePerGas: BigNumber;
+    maxPriorityFeePerGas: BigNumber;
+  };
+  retryNumber?: number;
+}): Promise<boolean> {
+  const blockProtection = await ethers.getContractAt('BlockProtection', '0xCC268041259904bB6ae2c84F9Db2D976BCEB43E5', params.wallet);
+  const targetBlockNumber = (await httpProvider.getBlockNumber()) + 2;
+
+  const populatedTx = await blockProtection.populateTransaction.callWithBlockProtection(
+    params.executeTx.to, // address _to,
+    params.executeTx.data, // bytes memory _data,
+    targetBlockNumber // uint256 _blockNumber
+  );
+
+  const signedTx = await params.wallet.signTransaction({
+    ...params.gasParams,
+    to: populatedTx.to!,
+    gasLimit: populatedTx.gasLimit!.toNumber(),
+    data: populatedTx.data!,
+  });
 
-    await sleep(DELAY);
+  console.log('[Execution] Sending transaction in block', targetBlockNumber);
+  // const protect = new ethers.providers.JsonRpcProvider('https://rpc.flashbots.net');
+  // const protectTx = await protect.sendTransaction(signedTx);
+  // console.log(`[Execution] Transaction submitted via protect rpc - https://protect.flashbots.net/tx/?hash=${protectTx.hash}`);
+
+  const bundle: FlashbotBundle = [
+    {
+      signedTransaction: signedTx,
+    },
+  ];
+  console.log('[Execution] Signed with hash:', signedTx);
+  if (await submitBundleForBlock(bundle, targetBlockNumber)) {
+    console.log('[Execution] Pending trade', params.pendingTrade._id, 'executed via', params.bestSetup.swapper);
+    return true;
+  }
+  if (!params.retryNumber) params.retryNumber = 0;
```

Comment:
```suggestion
  if (!params.retryNumber) return false;
```

Reply:
no return, if no params it means is the first call :)